gordon/interfaces.py

```diff
@@ -120,3 +120,46 @@ class IPublisherClient(IGenericPlugin):
         Args:
             event_msg (IEventMessage): Message ready to be sent to destination.
         """
+
+
+class IMetricRelay(Interface):
+    """Create and publish (push) metrics on demand."""
+
+    def incr(metric_name, value=1):
+        """Increment a metric. Create one if new."""
+
+    def timer(metric_name):
+        """Return a :class:`ITimerMetric`. Create one if new."""
+
+    def set_counter(metric_name, counter):
```

Comment:
I don't understand why this and `set_timer` are separate methods, couldn't `add_metric(metric)` be just as useful? I also disagree that the metric name should be stored with the relay rather than the metric itself